test/slug.test.js

```diff
@@ -50,9 +50,10 @@ describe('slug', function () {
     assert.throws(slug, /slug\(\) requires a string argument/)
   })
 
-  it('should replace whitespaces with replacement', function () {
+  it('should replace whitespaces with replacement and preserve default replacement', function () {
     assert.strictEqual(slug('foo bar baz'), 'foo-bar-baz')
     assert.strictEqual(slug('foo bar baz', '_'), 'foo_bar_baz')
+    assert.strictEqual(slug('foo-bar baz', '_'), 'foo-bar_baz')
```

Comment:
I don't think this is right. If you specify that you want your lone non-alphanumeric character to be `_`, then treating `-` as a preserved character is a bug. (We don't preserve `_`.) I could see *maybe* arguing that certain symbols (`-`, `_`, and a handful of others) be converted to the replacement character along with spaces. But preserving them seems like a bug to me.

Reply:
@Trott You are right!

I believe I meant to write :

```
    assert.strictEqual(slug('foo-bar baz', '_'), 'foo_bar_baz')
```

Reply:
`slug()` explicitly does not preserve punctuation/symbols by default. `slugify()` does.

```console
> slug('foo-bar_baz')
'foo-barbaz'
> slugify('foo-bar_baz')
'foo-bar_baz'
>
```

If you want slug to treat convert `-` to `_`, you need to add it to the charmap.

```console
> slug.charmap['-'] = '_'
'-'
> slug('foo-bar baz', {replacement: '_'})
'foo_bar_baz'
> 
```

Reply:
@Trott Thanks for your response ( :

I think I'll use this solution